src/ui/textfield.rs

```diff
@@ -41,11 +90,20 @@ impl TextField {
     /// Creates a new text field, with the optionally provided initial text.
     pub fn new(initial_text: Option<&str>) -> Self {
         let text_utf8: String = initial_text.unwrap_or("").into();
+        let length = text_utf8.len();
+
         Self {
             text: text_utf8.chars().collect(),
             text_utf8,
             focused: false,
             blink_start: 0.0,
+
+            selection: Selection {
+                first: length,
+                second: length,
+            },
```

Comment:
Using the UTF-8 length here is incorrect! Note that the selection positions are expressed using indices into the `text` table, whose length may be shorter than that of `text_utf8`.

As a note for the future, `text_utf8` is only used as a buffer for using the text in modules outside this one, as pretty much noone in Rust-land speaks `&[char]` for strings, however they do speak `&str`.